packages/flutter/lib/src/material/material_button.dart

```diff
@@ -84,34 +92,45 @@ abstract class MaterialButtonState<T extends MaterialButton> extends State<T> {
 
   void _handleHighlightChanged(bool value) {
     setState(() {
+      // mostly just used by the RaisedButton subclass to change the elevation
       highlight = value;
     });
   }
 
   Widget build(BuildContext context) {
-    Widget contents = new Container(
-      padding: new EdgeDims.symmetric(horizontal: 8.0),
-      child: new Center(
-        widthFactor: 1.0,
-        child: config.child
+    Widget contents = new InkWell(
+      onTap: config.enabled ? config.onPressed : null,
```

Comment:
You can skip the `config.enabled` check because `config.onPressed` will be null anyway if `config.enabled` is false.